core/rio/rdfxml/src/test/java/org/eclipse/rdf4j/rio/rdfxml/RDFXMLPrettyWriterTest.java

```diff
@@ -104,6 +109,58 @@ public void outOfSequenceItemsAreNotAbbreviated() throws RDFHandlerException, IO
 		assertEquals(Arrays.asList("<rdf:RDF", "<rdf:Bag", "<rdf:_0", "<rdf:_2"), rdfLines);
 	}
 
+	@Test
+	public void compactXMLPrintTest() throws RDFHandlerException, IOException, ClassNotFoundException,
+			NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+		OutputStream outputStream = new ByteArrayOutputStream();
+		PrintStream printStream = new PrintStream(outputStream);
+		System.setOut(printStream);
+
+		RDFWriter rdfXmlWriter = new RDFXMLPrettyWriter(System.out);
+
+		Class<?> arrangedWriterCls = Class.forName("org.eclipse.rdf4j.rio.turtle.ArrangedWriter");
+		Constructor<?> constructor = arrangedWriterCls.getConstructor(RDFWriter.class);
+		constructor.setAccessible(true);
+		RDFWriter arrangedXmlWriter = (RDFWriter) constructor.newInstance(rdfXmlWriter);
+
+		arrangedXmlWriter.set(BasicWriterSettings.PRETTY_PRINT, false);
+		arrangedXmlWriter.set(BasicWriterSettings.INLINE_BLANK_NODES, true);
+		arrangedXmlWriter.set(XMLWriterSettings.COMPACT_XML, true);
+
+		ValueFactory vf = SimpleValueFactory.getInstance();
+
+		IRI subject = vf.createIRI("http://example.org/subject");
+		IRI relation = vf.createIRI("http://example.org/relation");
+
+		LinkedHashModel model = new LinkedHashModel();
+		BNode bnode = vf.createBNode("bnode");
+
+		model.add(subject, relation, bnode);
+		model.add(bnode, RDFS.LABEL, vf.createLiteral("the bnode"));
+		model.setNamespace(RDFS.NS);
+		model.setNamespace("ex", "http://example.org/");
+
+		List<Statement> shuffledStatements = new ArrayList<>(model);
+		Collections.shuffle(shuffledStatements);
+		Rio.write(shuffledStatements, arrangedXmlWriter);
```

Comment:
Like I said above: I don't really think you need this shuffle on the order of statements here: your test should only be about the improvement to the RDFXMLPrettyWriter that you did. Making sure statements are always in the correct order is not part of that class' job. So by doing this shuffle, all you're really testing is that ArrangedWriter is doing its job - and that should not be part of _this_ unit test. 

The only thing you should be testing here is that _if_ statements are fed in in the correct order, and the COMPACT_XML setting is turned on, the output is as expected. So I'd suggest that instead of creating a Model and then shuffling that Model, you just create two Statement using the ValueFactory, then add those two statements to a simple ArrayList in the correct order, and then pass that Arraylist to the RDFXMLPrettyWriter.